src/Image/Image.php

```
@@ -54,6 +56,37 @@ public function convertImage($image, $fileName, $quality = 100)
         return $result;
     }
 
+    /**
+     * Get image extesnion
+     *
+     * @param string $fileName
+     *
+     * @return string
+     */
+    private function getImageExtension($fileName)
+    {
+        $exifImageLists = [
+            IMAGETYPE_GIF => 'gif',
+            IMAGETYPE_JPEG => 'jpg',
+            IMAGETYPE_BMP => 'bmp',
+            IMAGETYPE_PNG => 'png',
+        ];
+
+        if (function_exists('exif_imagetype') && file_exists($fileName)) {
+            $imageType = exif_imagetype($fileName);
+
+            if (!in_array($imageType, $exifImageLists)) {
+                $extension = strtolower(pathinfo($fileName, PATHINFO_EXTENSION));
+            } else {
+                $extension = $exifImageLists[$imageType];
+            }
+        } else {
+            $extension = strtolower(pathinfo($fileName, PATHINFO_EXTENSION));
```

I think we can check detected image extension should be existed in the `$exifImageLists`.

If not, we should throw `InvalidArgumentException`.

---

Throwing an exception for invalid (not supported) file types would be good.